aide_design/cdc_functions.py

```diff
@@ -91,28 +103,30 @@ def _flow_available(Diam, HeadlossCDC):
 
 # Length of tube required to get desired head loss at maximum flow based on 
 # the Hagen-Poiseuille equation.    
-@u.wraps(u.m, [u.m**3/u.s, u.m, u.m, u.m**2/u.s, None], False)
-def _len_tube(Flow, Diam, HeadLoss, Nu, KMinor):
-    num1 = pc.gravity.magnitude * HeadLoss * math.pi * (Diam**4)
-    denom1 = 128 * Nu * Flow
+@u.wraps(u.m, [u.m**3/u.s, u.m, u.m, u.kg/u.m**3, u.degK, None, None], False)
+def _len_tube(Flow, Diam, HeadLoss, conc_chem, temp, en_chem, KMinor):
+    """Length of tube required to get desired head loss at maximum flow based on 
+    the Hagen-Poiseuille equation."""
+    num1 = pc.gravity.magnitude * HeadLoss * np.pi * (Diam**4)
+    denom1 = 128 * viscosity_kinematic_chem(conc_chem, temp, en_chem) * Flow
     num2 = Flow * KMinor
-    denom2 = 16 * math.pi * Nu
+    denom2 = 16 * np.pi * viscosity_kinematic_chem(conc_chem, temp, en_chem)
     len = ((num1/denom1) - (num2/denom2))
-    return len
+    return len.magnitude
 
 
 
 #==============================================================================
 # Helper Functions     
 #==============================================================================
-@u.wraps(None, [u.m**3/u.s, u.kg/u.m**3, u.kg/u.m**3, u.m, u.m], False)
+@u.wraps(None, [u.m**3/u.s, u.kg/u.m**3, u.kg/u.m**3, u.m, u.m, None, None], False)
 def _n_tube_array(FlowPlant, ConcDoseMax, ConcStock, 
-                  DiamTubeAvail, HeadlossCDC): 
+                  DiamTubeAvail, HeadlossCDC, Ratio_Error, KMinor): 
     
     np.ceil((FlowPlant * ConcDoseMax
-               )/ ConcStock*_flow_available(DiamTubeAvail, HeadlossCDC).magnitude) 
+               )/ ConcStock*max_linear_flow(DiamTubeAvail, HeadlossCDC, Ratio_Error, KMinor).magnitude) 
```

Comment:
Why are there two calculations that are identical here? It might be better to put one calc into a variable and return that variable.